src/MatrixClient.ts

```diff
@@ -1354,15 +1355,14 @@ export class MatrixClient extends EventEmitter {
     }
 
     /**
-     * Creates a room. This does not break out the various options for creating a room
-     * due to the large number of possibilities. See the /createRoom endpoint in the
```

Comment:
Obviously, I've just ignored what you've said here and typed it anyway. I'd like to understand the reasoning behind this. I think the `/createRoom` endpoint is fairly stable these days and we're not likely to break someones obscure usage by explicitly typing things, and the value of having your editor provide you inline types and docs is an improvement over scanning the spec for the property keys, imo.

Reply:
argh, missed this in the review. Well, per the review comment, it's because I'd like it to be autogenned but also it's been 46 years or something so whatever.